Fast RTPS 1.7.2 adoption [4832] (#31)
eProsima/Micro-XRCE-DDS-docs: docs/agent.rst

------------

To run the *Agent* you should build it as indicated in :ref:`installation_label`.
Once it is built successfully, you just need to launch it executing the following command.

**UDP communication** ::

    $ ./MicroXRCEAgent --udp <local-port> [--discovery <discovery-port>] [--refs <refs-file>]

> Launch a UDP server over port ``<local-port``. It has the following options:

Reviewer: ```suggestion
Launch a UDP server over port ``<local-port>``. It has the following options:
```

Author: Done!

* ``--discovery <discovery-port>`` (only Linux): select a port for the discovery server which has the port number ``<7400>`` by default.
* ``--refs <refs-file>``: load a reference file using the relative path to the working directory.

>   This reference file shall be composed by a set of Fast RTPS profile following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ describes in Fast RTPS.

Reviewer: ```suggestion
  This reference file shall be composed by a set of Fast RTPS profiles following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ described in Fast RTPS.
```

Author: Done!

>   The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity, so it could be used by the *Clients* in order to create entities by reference.

Reviewer: ```suggestion
  The ``profile_name`` attribute of each profile represents a reference to an XRCE-Entity so that it could be used by the *Clients* to create entities by reference.
```

Author: Done!

**TCP communication** ::

    $ ./MicroXRCEAgent --tcp <local-port> [--discovery <discovery-port>] [--refs <refs-file>]

> Launch a TCP server over port ``<local-port``. It has the following options:

Reviewer: ```suggestion
Launch a TCP server over port ``<local-port>``. It has the following options:
```

Author: Done!

* ``--discovery <discovery-port>`` (only Linux): select a port for the discovery server which has the port number ``<7400>`` by default.
* ``--refs <refs-file>``: load a reference file using the relative path to the working directory.

>   This reference file shall be composed by a set of Fast RTPS profile following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ describes in Fast RTPS.

Reviewer: ```suggestion
  This reference file shall be composed by a set of Fast RTPS profiles following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ described in Fast RTPS.
```

Author: Done!

>   The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity, so it could be used by the *Clients* in order to create entities by reference.

Reviewer: ```suggestion
  The ``profile_name`` attribute of each profile represents a reference to an XRCE-Entity so that it could be used by the *Clients* to create entities by reference.
```

Author: Done!

**Serial communication** (only Linux) ::

    $ ./MicroXRCEAgent --serial <device-name> [--baudrate <baudrate>] [--refs <refs-file>]

> Launch a Serial server over device ``<device-name>>``. It has the following options:

Reviewer: ```suggestion
Launch a Serial server over device ``<device-name>``. It has the following options:
```

Author: Done!

* ``--baudrate <baudrate>``: set the baud rate of the communication. It can take the following values:
  0, 50, 75, 110, 134, 150, 200, 300, 600, 1200, 1800, 240, 4800, 9600, 19200, 38400, 57600, 115200 (default), 230400, 460800, 500000, 576000, 921600, 1000000, 1152000, 1500000, 2000000, 2500000, 3000000, 3500000 or 4000000 Bd.
* ``--refs <refs-file>``: load a reference file using the relative path to the working directory.

>   This reference file shall be composed by a set of Fast RTPS profile following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ describes in Fast RTPS.

Reviewer: ```suggestion
  This reference file shall be composed by a set of Fast RTPS profiles following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ described in Fast RTPS.
```

Author: Done!

>   The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity, so it could be used by the *Clients* in order to create entities by reference.

Reviewer: ```suggestion
  The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity so that it could be used by the *Clients* to create entities by reference.
```

Author: Done!

**Pseudo-Serial communication** (only Linux) ::

    $ ./MicroXRCEAgent --pseudo-serial [--baudrate <baudrate>] [--refs <refs-file>]

Launch a pseudo-serial server which provide a device to which *Clients* can connect. It has the following options:

* ``--baudrate <baudrate>``: set the baud rate of the communication. It can take the following values:
  0, 50, 75, 110, 134, 150, 200, 300, 600, 1200, 1800, 240, 4800, 9600, 19200, 38400, 57600, 115200 (default), 230400, 460800, 500000, 576000, 921600, 1000000, 1152000, 1500000, 2000000, 2500000, 3000000, 3500000 or 4000000 Bd.
* ``--refs <refs-file>``: load a reference file using the relative path to the working directory.

>   This reference file shall be composed by a set of Fast RTPS profile following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ describes in Fast RTPS.

Reviewer: ```suggestion
  This reference file shall be composed by a set of Fast RTPS profiles following the `XML syntax <https://eprosima-fast-rtps.readthedocs.io/en/latest/xmlprofiles.html>`_ described in Fast RTPS.
```

Author: Done!

>   The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity, so it could be used by the *Clients* in order to create entities by reference.

Reviewer: ```suggestion
  The ``profile_name`` attribute of each profile represents a reference for an XRCE-Entity so that it could be used by the *Clients* to create entities by reference.
```

Author: Done!
